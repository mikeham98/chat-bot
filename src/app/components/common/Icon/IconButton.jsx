import React from 'react';
import styles from '../../../../themes/components/icon.scss';
import Icon from "./index";

const IconButton = ({onClick, type, icon, style, color, hoverColor, height, width, disabled}) => {
    return (
        <button
            type={type}
            onClick={onClick}
            disabled={disabled}
            style={style}
            className={styles.iconButton}
        >
            <Icon
                IconComponent={icon}
                color={color}
                hoverColor={hoverColor}
                height={height}
                width={width}
            />
        </button>
    );
};

export default React.memo(IconButton);