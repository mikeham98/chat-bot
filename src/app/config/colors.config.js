const flatRed = '#f4244e';
const flatOrange = '#f87b2b';
const flatYellow = '#F1C40E';
const flatGreen = '#30CB71';
const flatTeal = '#1ccb9e';
const flatBlue = '#3598DA';
const flatLavender = '#6872FF';
const flatPurple = '#9A58B9';

export const flatColours = [
    flatRed,
    flatOrange,
    flatYellow,
    flatGreen,
    flatTeal,
    flatBlue,
    flatLavender,
    flatPurple
];