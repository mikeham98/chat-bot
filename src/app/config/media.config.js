export const gif = 'gif';
export const img = 'img';
export const link = 'link';