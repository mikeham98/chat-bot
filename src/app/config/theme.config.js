export const lightTheme = 'light';
export const darkTheme = 'dark';

export const oppositeTheme = (theme) => {
    if(theme === lightTheme) {
        return darkTheme;
    }
    return lightTheme;
};