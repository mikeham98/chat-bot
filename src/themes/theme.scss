@import 'colors';

.darkTheme {
  color: white;
  background-color: $flat-black;
  .messageOptionWrapper .messageOption, .leftMessage .message {
    border-color: $jet-black;
  }
  .messageOptionWrapper .messageOptionDisabled, .messageOptionWrapper .messageOptionDisabled:hover {
    color: $dark-gray;
  }

  .mediaLinkPreviewWrapper .mediaLinkPreviewMeta .mediaLinkPreviewDescription, .leftMessage .message {
    color: white;
  }
  .conversationWrapper,
  .sideBarMenu,
  .mediaLinkPreviewWrapper,
  .mediaImagePreviewImage,
  .headerLayout {
    border-color: $dark-gray;
  }
  .mediaLinkPreviewWrapper .mediaLinkPreviewMeta, .leftMessage .message {
    background-color: $jet-black;
  }
  .messageFormWrapper, .headerLayout, .sideBarMenu, .messageOptionWrapper, .messageContainer {
    background-color: $flat-black;
  }
  .messageFormWrapper {
    input {
      color: white
    }
  }
}