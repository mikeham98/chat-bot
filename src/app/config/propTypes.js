import PropTypes from 'prop-types';
import {gif, img, link} from "./media.config";

export const OptionPropTypes = PropTypes.shape({
    id: PropTypes.number.isRequired,
    option: PropTypes.string.isRequired,
});

export const MediaObjectPropTypes = PropTypes.shape({
    id: PropTypes.number.isRequired,
    src: PropTypes.string.isRequired,
    type: PropTypes.oneOf([gif, img, link]).isRequired,
    imagePreview: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
});

export const MediaPropTypes = PropTypes.arrayOf(MediaPropTypes);

export const OptionsPropTypes = PropTypes.arrayOf(OptionPropTypes);

export const MessagesPropTypes = PropTypes.arrayOf(PropTypes.shape({
    content: PropTypes.shape({
        body: PropTypes.string.isRequired,
        media: MediaPropTypes,
        options: OptionsPropTypes,
    }),
    recipient: PropTypes.shape({
        userId: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        userName: PropTypes.string
    }),
    createdAt: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.instanceOf(Date)
    ]).isRequired,
    createdBy: PropTypes.shape({
        userId: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        userName: PropTypes.string,
    }).isRequired
}));