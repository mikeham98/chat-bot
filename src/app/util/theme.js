import {darkTheme, lightTheme} from '../config/theme.config';

export const oppositeTheme = (theme) => {
    if(theme === lightTheme) {
        return darkTheme;
    }
    return lightTheme;
};