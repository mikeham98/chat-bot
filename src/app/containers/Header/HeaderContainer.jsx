import React from 'react';
import styles from '../../../themes/components/header.scss';
import {getTheme, setTheme} from '../../actions/settings/index.actions';
import IconButton from '../../components/common/Icon/IconButton';
import moonIcon from '../../../../assets/icons/moon.svg';
import sunIcon from '../../../../assets/icons/sun.svg';
import {connect} from 'react-redux';
import {lightTheme, oppositeTheme} from "../../config/theme.config";

export class HeaderContainer extends React.PureComponent {
    constructor(props) {
        super(props);
        this.handleChangeTheme = this.handleChangeTheme.bind(this);
    }

    componentDidMount() {
        this.props.getTheme();
    }

    isLightTheme() {
        return this.props.theme === lightTheme;
    }

    handleChangeTheme() {
        const newTheme = oppositeTheme(this.props.theme);
        this.props.setTheme(newTheme);
    }

    render() {
        const {name} = this.props;
        return (
            <div className={styles.headerWrapper}>
                <h2>{name}</h2>
                <div className={styles.themeSwitch}>
                    <IconButton
                        icon={this.isLightTheme() ? moonIcon : sunIcon}
                        onClick={this.handleChangeTheme}
                        color={this.isLightTheme() ? '#172B4D' : 'white'}
                        height='30px'
                        width='30px'
                    />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    let conversations = state.conversations.conversations;
    const currentConversation = conversations.find(conversation => conversation.id === state.conversations.currentConversationId);
    return {
        name: currentConversation && currentConversation.profile.name,
        theme: state.settings.theme
    }
};

export default connect(mapStateToProps, {getTheme, setTheme})(HeaderContainer)