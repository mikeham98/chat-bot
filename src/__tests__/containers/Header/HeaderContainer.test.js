import React from 'react';
import {shallow} from 'enzyme';
import {HeaderContainer} from '../../../app/containers/Header/HeaderContainer';
import IconButton from '../../../app/components/common/Icon/IconButton';
import {darkTheme, lightTheme} from "../../../app/config/theme.config";

describe('HeaderContainer', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<HeaderContainer {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            theme: 'light',
            getTheme: jest.fn(),
            setTheme: jest.fn(),
            setColor: jest.fn(),
            currentConversation: {
                id: "3",
                userId: "bot42",
                userName: "travel_bot",
                color: "#9A58B9",
                profile: {
                    name: "TravelBot",
                    img: "https://images.unsplash.com/photo-1496046744122-2328018d60b6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1664&q=80"
                },
                read: true,
                latest: {
                    message: "So just to confirm, you want to go to Florida for 2 weeks from August 12th (fly out 10am) August 26 (return flight 2pm) with 3 people.",
                    createdAt: "2019-07-26T20:36:26.725Z"
                }
            },
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a light theme with a h2 with the name and a ColorIcon', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a dark theme with a h2 with the name and a ColorIcon', () => {
            props.theme = darkTheme;
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a light theme with a h2 with the name and a ColorIcon with ColorPicker component', () => {
            wrapper().find('ColorIcon').first().props().onClick();
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('props', () => {
            describe('getTheme', () => {
                it('should be called on mount', () => {
                    expect(props.getTheme).not.toHaveBeenCalled();
                    wrapper();
                    expect(props.getTheme).toHaveBeenCalledTimes(1);
                });
            });
            describe('setTheme', () => {
                it('should be called on click of IconButton and pass in dark theme', () => {
                    expect(props.setTheme).not.toHaveBeenCalled();
                    wrapper().find(IconButton).first().props().onClick();
                    expect(props.setTheme).toHaveBeenCalledWith(darkTheme);
                });
                it('should be called on click of IconButton and pass in light theme', () => {
                    props.theme = darkTheme;
                    expect(props.setTheme).not.toHaveBeenCalled();
                    wrapper().find(IconButton).first().props().onClick();
                    expect(props.setTheme).toHaveBeenCalledWith(lightTheme);
                });
                it('should be called with red and 3 onChange of ColorPicker', () => {
                    wrapper().find('ColorIcon').first().props().onClick();
                    expect(props.setColor).not.toHaveBeenCalled();
                    wrapper().find('ColorPicker').first().props().onChange('red');
                    expect(props.setColor).toHaveBeenCalledWith('red', '3');
                });
            });
        });
    });
});