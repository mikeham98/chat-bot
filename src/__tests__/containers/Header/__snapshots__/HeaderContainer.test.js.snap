// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer render should return a dark theme with a h2 with the name and a ColorIcon 1`] = `
<div
  className="headerWrapper"
>
  <h2>
    TravelBot
  </h2>
  <div
    className="headerThemeToggle"
  >
    <Memo(IconButton)
      color="white"
      height="30px"
      icon={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      onClick={[Function]}
      width="30px"
    />
  </div>
  <div
    className="headerColorPicker"
  >
    <ColorIcon
      color="#9A58B9"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`HeaderContainer render should return a light theme with a h2 with the name and a ColorIcon 1`] = `
<div
  className="headerWrapper"
>
  <h2>
    TravelBot
  </h2>
  <div
    className="headerThemeToggle"
  >
    <Memo(IconButton)
      color="#172B4D"
      height="30px"
      icon={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      onClick={[Function]}
      width="30px"
    />
  </div>
  <div
    className="headerColorPicker"
  >
    <ColorIcon
      color="#9A58B9"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`HeaderContainer render should return a light theme with a h2 with the name and a ColorIcon with ColorPicker component 1`] = `
<div
  className="headerWrapper"
>
  <h2>
    TravelBot
  </h2>
  <div
    className="headerThemeToggle"
  >
    <Memo(IconButton)
      color="#172B4D"
      height="30px"
      icon={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      onClick={[Function]}
      width="30px"
    />
  </div>
  <div
    className="headerColorPicker"
  >
    <ColorIcon
      color="#9A58B9"
      onClick={[Function]}
    />
  </div>
  <ColorPicker
    onChange={[Function]}
  />
</div>
`;
